shared/actions/route-tree.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import * as I from 'immutable'
+import * as Constants from '../constants/route-tree'
+import {RouteNode} from '../route-tree'
+import type {Action} from '../constants/types/flux'
+
+// Set the route tree
+export function setRoutes (routeTree: typeof RouteNode): Action {
+  return {
+    type: Constants.setRoute,
+    payload: {routeTree},
+  }
+}
+
+// Switch to a new parent path, keeping the subpath. E.g.:
+// switchTo('settings') will navigate to settings tab and whatever subpath was
+// previously selected
+export function switchTo (...path): Action {
+  return {
+    type: Constants.switchTo,
+    payload: {path},
+  }
+}
+
+//TODO: check type of path objects to ensure selected key
+
+// Navigate to a new absolute path.
+// You can specify paths as either strings:
+//   navigateTo('foo', 'bar')
+// Or objects with route props:
+//   navigateTo({selected: 'foo', prop1: 'hello'}, {selected: 'bar', prop2: 'world'})
```

Comment:
This comment is confusing, is prop1 a prop that will be passed to the component? or is it something that should only be in a RouteNode (?) type?